packages/name-controller/src/NameController.ts

```diff
@@ -441,6 +442,14 @@ export class NameController extends BaseController<
     const normalizedValue = this.#normalizeValue(value, type);
     const normalizedVariation = this.#normalizeVariation(variationKey, type);
 
+    if (
+      ![normalizedValue, normalizedVariation].every((key) =>
+        isSafeDynamicKey(key),
```

Comment:
Nit: This may be (slightly) more readable:

``` suggestion
      [normalizedValue, normalizedVariation].some((key) =>
        !isSafeDynamicKey(key),

Reply:
Also, should we add a test for this? I think that would bring us back up to 100%.

Reply:
Added tests for all three files here: https://github.com/MetaMask/core/pull/4041/commits/a145237447a84bf3fa534ad0bc54cd7b22601a37